web_ui_next.py

```diff
@@ -8,34 +8,38 @@
 
 SYSTEM_MESSAGE_KEY = "agent_system_message"
 HISTORY_NAME_KEY = "history_name"
+AGENT_NAME_KEY = "agent_name"
 
 
-@st.cache_resource
-def get_agent_store():
-    return AgentStore(
-        "agents",
-        name_to_plugin_class={
-            "PlaywrightPlugin": PlaywrightPlugin,
-        },
-    )
+agent_store = AgentStore(
+    "agents",
+    name_to_plugin_class={
+        "PlaywrightPlugin": PlaywrightPlugin,
+    },
+)
 
 
-agent_store = get_agent_store()
+sidebar = st.sidebar
 
 
-@st.cache_resource
-def get_agent(agent_name):
-    new_agent = agent_store.load_agent(
-        agent_name, default_kwargs={"plugins": {"PlaywrightPlugin": PlaywrightPlugin()}}
+def load_agent():
```

Comment:
Agent does need to be cached, but we can do it ourselves in session_state. That way we have more control over the logic (e.g. we can implement a "Reload Agent" button.